src/DankCord/Objects.py

```diff
@@ -35,62 +37,112 @@ def __init__(self, data: dict) -> None:
         except Exception as e:
             self.embeds = []
             raise e
-
-        self.author: Author = Author(data["author"])
         self.channel: int = int(data["channel_id"])
         self.timestamp: int = data["timestamp"]
-
         try:
             self.components: list = []
+            self.buttons: list = []
+            self.dropdowns: list = []
             actionrows = data["components"]
             actionrow_objects = []
             for i in actionrows:
-                actionrow_objects.append(ActionRow(i))
+                actionrow_objects.append(ActionRow(i, self.id, client))
             self.components = actionrow_objects
+            button_objects = []
+            dropdown_objects = []
+            for i in self.components:
+              for i2 in i.components:
+                if isinstance(i2, Button):
+                  button_objects.append(i2)
+                if isinstance(i2, Dropdown):
+                  dropdown_objects.append(i2)
+            self.buttons = button_objects
+            self.dropdowns = dropdown_objects
         except Exception as e:
             self.components: list = []
             raise e
 
+    def fetch_button(self, label: str):
+        returned_button = None
+        for i in self.buttons:
```

Comment:
List comprehension could make this much better

Reply:
done